lib/betterdocs/generator/swagger.rb

```diff
@@ -0,0 +1,391 @@
+        when 'object'
+          res = get_schema_ref(sub_cls) if sub_cls
+          res[:description] = description
+        end
+        res[:nullable] = true if nullable
+        res
+      end
+
+      def get_dto_name(representer)
+        return representer.name.demodulize if representer&.name
+
+        nil
+      end
+
+      def add_links_definition(definitions, cls, value)
+        definition = initialise_definition(definitions, cls)
+        unless definition[:properties][:links]
+          definition[:properties][:links] = {
+            type: 'array',
+            nullable: false,
+            items: { type: 'object', properties: {
+              rel: { type: 'string', enum: [] },
+              href: { type: 'string' }
+            }, required: %w[rel href] }
+          }
+        end
+        enum = definition[:properties][:links][:items][:properties][:rel][:enum]
+        enum.push(value) unless enum.include?(value)
+      end
+
+      def initialise_definition(definitions, cls)
+        definitions[cls] = { type: 'object', properties: {} } if definitions[cls].nil?
+        definitions[cls]
+      end
+
+      def add_response_schema(definitions, response)
+        return nil if response.full?.nil?
+
+        subs = {}
+        resp_cls = get_dto_name(response.representer)
+        (response.full?(:properties) || []).each do |p|
+          subs[p.full_name] = p.sub_representer? if p.sub_representer?
+          sub_cls = get_dto_name(subs[p.nesting_name])
+          cls = sub_cls || resp_cls
+          definition = initialise_definition(definitions, cls)
+          definition[:properties][p.public_name] = get_schema(p.types, get_dto_name(p.sub_representer?), p.description)
+        end
+        (response.full?(:links) || []).each do |l|
+          sub_cls = get_dto_name(subs[l.nesting_name])
+          add_links_definition(definitions, sub_cls || resp_cls, l.public_name)
+        end
+        get_schema_ref(resp_cls)
+      end
+
+      def get_schema_ref(name)
+        {
+          "$ref": "#/components/schemas/#{name}"
+        }
+      end
+
+      def add_required_to_definitions(definitions)
+        definitions.each do |def_key, d|
+          req = []
+          d[:properties].each do |key, p|
+            req.push(key) if p.key?(:nullable)
+          end
+          definitions[def_key][:required] = req unless req.length.zero?
+        end
+      end
+
+      def add_body(definitions, params, action, name)
+        if action.json_params.full?
+          payload_name = "#{name.titleize.gsub(/\s/, '')}Request"
+          definitions[payload_name] = get_request_definition(action.json_params.values)
+          params[:requestBody] = wrap_content_object({ schema: get_schema_ref(payload_name) }, "#{name} Request")
+        end
+      end
+
+      def add_path_params(params, from_path)
+        from_path.each do |param|
+          params = add_param(params, param[:name], 'path', true, param[:type], param[:name])
```

Comment:
```suggestion
          add_param(params, param[:name], 'path', true, param[:type], param[:name])
```
assignment is unused